lib/tap.js

```diff
@@ -20,11 +20,12 @@ const rootDomain = new Domain((er, type) => {
   }
 })
 
-// We test that file separately, and removing stdout
-// causes some weird behavior in the Node test runner.
 /* istanbul ignore next */
 if (!process.stdout) {
-  require('./stdio-polyfill')()
+  // Set exitCode in case process.stderr is also invalid
+  process.exitCode = 1
+  console.error('Output stream is invalid')
```

Comment:
@isaacs let me know if you want this message to be different or if you have any suggestion for how to test this.

Reply:
I guess it could be done with a child process, but honestly, it's such a weird scenario that this is probably fine.